pkg/config/lima_config_applier.go

```diff
@@ -96,6 +121,15 @@ func (lca *limaConfigApplier) Apply(isInit bool) error {
 		limaCfg = *cfgAfterInit
 	}
 
+	var sociEnabled bool
+	if lca.cfg.Snapshotter == nil {
+		sociEnabled = false
+	} else {
+		sociEnabled = (*lca.cfg.Snapshotter == "soci")
+	}
+
```

Comment:
we should add an error here if `snapshotter` doesn't have a valid value so users don't wonder why their config change isn't working